pkg/controller/docdb/dbcluster/dbcluster.go

```diff
@@ -0,0 +1,304 @@
+		(cr.Spec.ForProvider.DBClusterParameterGroupName != nil && !awsclient.IsEqualString(cr.Spec.ForProvider.DBClusterParameterGroupName, cluster.DBClusterParameterGroup)) ||
+		(cr.Spec.ForProvider.DBSubnetGroupName != nil && !awsclient.IsEqualString(cr.Spec.ForProvider.DBSubnetGroupName, cluster.DBSubnetGroup)) ||
+		(cr.Spec.ForProvider.DeletionProtection != nil && !awsclient.IsEqualBool(cr.Spec.ForProvider.DeletionProtection, cluster.DeletionProtection)) ||
+		(cr.Spec.ForProvider.EnableCloudwatchLogsExports != nil && !areSameElements(cr.Spec.ForProvider.EnableCloudwatchLogsExports, cluster.EnabledCloudwatchLogsExports)) ||
+		(cr.Spec.ForProvider.Port != nil && !awsclient.IsEqualInt64(cr.Spec.ForProvider.Port, cluster.Port)) ||
+		(cr.Spec.ForProvider.PreferredBackupWindow != nil && !awsclient.IsEqualString(cr.Spec.ForProvider.PreferredBackupWindow, cluster.PreferredBackupWindow)) ||
+		(cr.Spec.ForProvider.PreferredMaintenanceWindow != nil && !awsclient.IsEqualString(cr.Spec.ForProvider.PreferredMaintenanceWindow, cluster.PreferredMaintenanceWindow)) {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func (e *hooks) preUpdate(ctx context.Context, cr *svcapitypes.DBCluster, obj *svcsdk.ModifyDBClusterInput) error {
+	obj.DBClusterIdentifier = awsclient.String(meta.GetExternalName(cr))
+
+	if !awsclient.IsEqualString(cr.Spec.ForProvider.DBClusterIdentifier, obj.DBClusterIdentifier) {
+		obj.NewDBClusterIdentifier = cr.Spec.ForProvider.DBClusterIdentifier
+	}
+
+	obj.BackupRetentionPeriod = cr.Spec.ForProvider.BackupRetentionPeriod
+	obj.DBClusterParameterGroupName = cr.Spec.ForProvider.DBClusterParameterGroupName
+	obj.DeletionProtection = cr.Spec.ForProvider.DeletionProtection
+	obj.CloudwatchLogsExportConfiguration = generateCloudWatchExportConfiguration(
+		cr.Spec.ForProvider.EnableCloudwatchLogsExports,
+		cr.Status.AtProvider.EnabledCloudwatchLogsExports)
+	obj.EngineVersion = cr.Spec.ForProvider.EngineVersion
+	obj.MasterUserPassword = cr.Spec.ForProvider.MasterUserPassword
+	obj.Port = cr.Spec.ForProvider.Port
+	obj.PreferredBackupWindow = cr.Spec.ForProvider.PreferredBackupWindow
+	obj.PreferredMaintenanceWindow = cr.Spec.ForProvider.PreferredMaintenanceWindow
+	obj.VpcSecurityGroupIds = cr.Spec.ForProvider.VPCSecurityGroupIDs
+
+	// Should apply immediately?
+	obj.ApplyImmediately = awsclient.Bool(true)
+
+	return nil
+}
+
+func (e *hooks) postUpdate(ctx context.Context, cr *svcapitypes.DBCluster, resp *svcsdk.ModifyDBClusterOutput, upd managed.ExternalUpdate, err error) (managed.ExternalUpdate, error) {
+	if err != nil {
+		return managed.ExternalUpdate{}, err
+	}
+
+	meta.SetExternalName(cr, awsclient.StringValue(resp.DBCluster.DBClusterIdentifier))
+
+	return upd, nil
+}
+
+func (e *hooks) preCreate(ctx context.Context, cr *svcapitypes.DBCluster, obj *svcsdk.CreateDBClusterInput) error {
+	obj.AvailabilityZones = cr.Spec.ForProvider.AvailabilityZones
```

Comment:
Most of the assignments here take place in `GenerateCreateDBClusterInput` already and it's called before `preCreate`.

Reply:
Removed